### PR TITLE
Implementation of System ports initialization, Interface & Neighbor Synchronization...

### PR DESCRIPTION
**Pull request by minionatwork**
This PR includes implementation of System ports initialization & orchestration, Interface & Neighbor Synchronization using GLOBAL_APP_DB, Inband Interface Configuration.

HLD: https://github.com/Azure/SONiC/pull/639

### CODE REVIEW

File: common/table.cpp [hunk 1]
```diff
@@ -29,7 +29,8 @@ const TableNameSeparatorMap TableBase::tableNameSeparatorMap = {
    { STATE_DB,            TABLE_NAME_SEPARATOR_VBAR  },
    { GB_ASIC_DB,          TABLE_NAME_SEPARATOR_VBAR  },
    { GB_COUNTERS_DB,      TABLE_NAME_SEPARATOR_VBAR  },
-   { GB_FLEX_COUNTER_DB,  TABLE_NAME_SEPARATOR_VBAR  }
+   { GB_FLEX_COUNTER_DB,  TABLE_NAME_SEPARATOR_VBAR  },
+   { GLOBAL_APP_DB,       TABLE_NAME_SEPARATOR_COLON }
```
Comment (lguohan) [inline common/table.cpp:33]: better to use VBAR

File: common/schema.h [hunk 2]
```diff
@@ -19,6 +19,7 @@ namespace swss {
 #define GB_ASIC_DB      8
 #define GB_COUNTERS_DB  9
 #define GB_FLEX_COUNTER_DB  10
+#define GLOBAL_APP_DB   11
```
Comment (vganesan-nokia) [inline common/schema.h:22]: GLOBAL_APP_DB will be changed to CHASSIS_APP_DB

File: common/schema.h [hunk 3]
```diff
@@ -267,6 +270,12 @@ namespace swss {
 #define CFG_FG_NHG_PREFIX                           "FG_NHG_PREFIX"
 #define CFG_FG_NHG_MEMBER                           "FG_NHG_MEMBER"
 
+#define CFG_SYSTEM_PORT_TABLE_NAME                  "SYSTEM_PORT"
+#define CFG_VOQ_INBAND_INTERFACE_TABLE_NAME         "VOQ_INBAND_INTERFACE"
+
+#define VOQ_SYSTEM_INTERFACE_TABLE_NAME             "SYSTEM_INTERFACE"
+#define VOQ_SYSTEM_NEIGH_TABLE_NAME                 "SYSTEM_NEIGH"
```
Comment (vganesan-nokia) [inline common/schema.h:277]: Change VOQ to CHASSIS_APP to refer to the DB

File: common/table.cpp (hunk 1 above)
Comment (vganesan-nokia) [reply common/table.cpp:33]: o.k. We'll change to use VBAR.

Comment (vganesan-nokia) [reply common/table.cpp]: Fixed

File: common/schema.h (hunk 2 above)
Comment (vganesan-nokia) [reply common/schema.h]: Fixed

File: common/schema.h (hunk 3 above)
Comment (vganesan-nokia) [reply common/schema.h]: Fixed

File: common/schema.h [hunk 4]
```diff
@@ -21,6 +21,10 @@ namespace swss {
 #define GB_COUNTERS_DB  10
 #define GB_FLEX_COUNTER_DB  11
 
+//For chassis app db in supervisor/control card redis in chassis systems.
+//This DB ID is not to be used in local redis.
+#define CHASSIS_APP_DB  8
```
Comment (BrynXu) [inline common/schema.h:26]: due to issue [5351](https://github.com/Azure/sonic-buildimage/issues/5351), the db id has to be unique even they are used in different redis server. In Arista PR 5283, it is set to 12, we need to be in sync the value.

Comment (vganesan-nokia) [reply common/schema.h:26]: o.k. I'll set to CHASSIS_APP_DB id to 12.

Comment (vganesan-nokia) [reply common/schema.h]: Fixed

**Review by eswaranb — APPROVED** [review]

**Review by eswaranb — APPROVED** [review]

**Review by lguohan — APPROVED** [review]

File: common/database_config.json [hunk 5]
```diff
@@ -4,6 +4,11 @@
             "hostname" : "127.0.0.1",
             "port" : 6379,
             "unix_socket_path" : "/var/run/redis/redis.sock"
+        },
+        "redis_chassis":{
+            "hostname" : "redis_chassis.server",
+            "port" : 6380,
+            "unix_socket_path" : "/var/run/redis/redis_chassis.sock"
```
Comment (qiluo-msft) [inline common/database_config.json:11]: Please move chassis specific configuration out of the default database_config.json. The file here is to maintain backward-compatiblity to before multi-DB feature and minimal configuration for any sonic system.

For example, this file does not work with sonic-utilities.

If your use case need customization, just customize it and overwrite it in your use case only. Do not break others.

**Pull request merged**